fpmsyncd/bgpmon.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python
+
+""""
+Description: bgpmon.py -- populating bgp related information in stateDB.
+    script is started by supervisord in bgp docker when the docker is started.
+
+    Initial creation of this daemon is to assist SNMP agent in obtaining the 
+    BGP related information for its MIB support. The MIB that this daemon is
+    assiting is for the CiscoBgp4MIB (Neighbor state only). If there are other
+    BGP related items that needs to be updated in a periodic manner in the 
+    future, then more can be added into this process.
+
+    The script check if there are any bgp activities by monitoring the bgp
+    frr.log file timestamp.  If activity is detected, then it will request bgp
+    neighbor state via vtysh cli interface. This bgp activity monitoring is 
+    done periodically (every 15 second). When triggered, it looks specifically
+    for the neighbor state in the json output of show ip bgp neighbors json
+    and update the state DB for each neighbor accordingly.
+    In order to not disturb and hold on to the State DB access too long and
+    removal of the stale neighbors (neighbors that was there previously on 
+    previous get request but no longer there in the current get request), a
+    "previous" neighbor dictionary will be kept and used to determine if there
+    is a need to perform update or the peer is stale to be removed from the
+    state DB
+"""
+
+import commands
+import json
+import os
+import syslog
+import swsssdk
+import time
+import traceback
+
+class BgpStateGet():
+    def __init__(self):
+        self.ipv4_n = []
```

Comment:
>_n [](start = 17, length = 2)

What does `_n` mean?

Reply:
@qiluo-msft Originally it was named as "self.ipv4_neighbor".  It is a list to keep all the ipv4 neighbor's ip address. But I chose to shorten the name because the lines are getting too long and require multiple lines to wrap in order to allow user to still read it with a small width terminal (trying to keep everything within 79 chars...).  So I made a decision to chop the name to shorten it.  I have added more comments in the code to explain each variable usage as following:
        # list ipv4_n stores the IPV4 Neighbor peer Ip address
        # dic ipv4_n_state stores the IPV4 Neighbor peer state entries
        # list ipv6_n stores the IPV4 Neighbor peer Ip address
        # dic ipv6_n_state stores the IPV4 Neighbor peer state entries
        # list new_ipv4_n stores the new snapshot of IPV4 Neighbor ip address
        # dic new_ipv4_n_state stores the new snapshot of IPV4 Neighbor states 
        # list new_ipv6_n stores the new snapshot of IPV6 Neighbor ip address
        # dic new_ipv6_n_state stores the new snapshot of IPV4 Neighbor state
Hope this helps...